tests/test_scenario/test_hitless_upgrade.py

```diff
@@ -608,6 +663,261 @@ def test_new_connection_handling_during_migrating_and_moving(
         self._validate_default_state(client, expected_matching_conns_count=3)
         bind_thread.join()
 
+    @pytest.mark.timeout(300)  # 5 minutes timeout
+    def test_old_connection_shutdown_during_moving(
+        self,
+        fault_injector_client: FaultInjectorClient,
+        endpoints_config: Dict[str, Any],
+    ):
+        # it is better to use ip for this test - enables validation that
+        # the connection is disconnected from the original address
+        # and connected to the new address
+        endpoint_type = EndpointType.EXTERNAL_IP
+        logging.info("Testing old connection shutdown during MOVING")
+        client = _get_client_maint_events(
+            endpoints_config=endpoints_config, endpoint_type=endpoint_type
+        )
+
+        logging.info("Starting migration ...")
+        migrate_thread = Thread(
+            target=self._execute_migration,
+            name="migrate_thread",
+            args=(
+                fault_injector_client,
+                endpoints_config,
+                self.target_node,
+                self.empty_node,
+            ),
+        )
+        migrate_thread.start()
+
+        logging.info("Waiting for MIGRATING push notifications...")
+        ClientValidations.wait_push_notification(client, timeout=MIGRATE_TIMEOUT)
+        self._validate_maintenance_state(client, expected_matching_conns_count=1)
+
+        logging.info("Waiting for MIGRATED push notification ...")
+        ClientValidations.wait_push_notification(client, timeout=MIGRATE_TIMEOUT)
+        self._validate_default_state(client, expected_matching_conns_count=1)
+        migrate_thread.join()
+
+        moving_event = threading.Event()
+
+        def execute_commands(moving_event: threading.Event, errors: Queue):
+            while not moving_event.is_set():
+                try:
+                    client.set("key", "value")
+                    client.get("key")
+                except Exception as e:
+                    errors.put(
+                        f"Command failed in thread {threading.current_thread().name}: {e}"
+                    )
+
+        logging.info("Starting rebind...")
+        bind_thread = Thread(
+            target=self._execute_bind,
+            name="bind_thread",
+            args=(fault_injector_client, endpoints_config, self.endpoint_id),
+        )
+        bind_thread.start()
+
+        errors = Queue()
+        commands_threads = []
+        threads_count = 10
+        logging.info(f"Starting {threads_count} command execution threads...")
+        for _ in range(threads_count):
```

Comment:
You can use `ThreadPoolExecutor` to run multiple threads, this is more standard way to invoke such operations